Training/Training/Training.c

```diff
@@ -2,13 +2,50 @@
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
 // ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
+// Training.c
+// Program to check if a given number is prime or not.
+// Sudarson S
 // ------------------------------------------------------------------------------------------------
 
+#include<stdbool.h>
 #include<stdio.h>
+#include<stdlib.h>
+#include<string.h> // USED
+#include<errno.h>
+
+#define PRIME -1
+#define COMPOSITE -2
+#define NEITHER -3
+
+/// <summary>Returns user inputs only if they are valid positive integers</summary>
+static int GetValidInteger (char* prompt) {
+   bool output;
+   int validInt;
+   do {
+      char numberStr[4096];
+      printf ("%s", prompt);
+      fgets (numberStr, sizeof (numberStr), stdin);
+      if (output = strchr (numberStr, '\n') != NULL) {
+         errno = 0;
+         char* endptr = NULL;
+         validInt = strtol (numberStr, &endptr, 10);
+         output = endptr != numberStr && *endptr == '\n' && errno != ERANGE && validInt > 0;
+      }
+   } while (!output);
+   return validInt;
+}
+
+/// <summary>Returns if a number is prime, composite or neither</summary>
+int PrimeChecker (int num) {
+   int factors = 0;
+   for (int i = 1; i <= num; i++)
+      if (num % i == 0) factors++;
+   return factors == 2 ? PRIME : factors == 1 ? NEITHER : COMPOSITE;
```

Comment:
This logic can be improved. Instead of checking factors == 1, you can just check for num and return NEITHER. 
Moreover, you are waiting for the loop to go through each number between 1 and num before returning it is prime or composite. If the number is 12345, it'll check for 12346 numbers before it returns false. This should not be the case.